doc/Language/classtut.pod6

```diff
@@ -5,86 +5,60 @@
 =SUBTITLE A tutorial about creating and using classes in Raku
 
 X<|OOP>
+Raku provides a rich built-in syntax for defining and using classes. It makes writing classes
+expressive and short for most cases, but also provides mechanisms to cover
+the rare corner cases.
 
-Raku has a rich built-in syntax for defining and using classes.
+=head1 A quick overview
 
-A default constructor allows the setting of attributes for the created
-object:
+Let's start with an example to give an overview:
 
 =begin code
-class Point {
-    has Int $.x;
-    has Int $.y;
-}
-
 class Rectangle {
-    has Point $.lower;
-    has Point $.upper;
+    has Int $.a = 1;
+    has Int $.b = 1;
+
+    submethod TWEAK(:$log) {
+        say "Constructing Rectangle ($!a|$!b)" if $log;
+    }
 
     method area(--> Int) {
```

Comment:
If you *really* want to introduce TWEAK, this area could be computed there, and of course $!a and $!b should be private in order to make the object immutable.

Reply:
I think introducing public and private attributes has precedence over TWEAK in the very first example. I still imagine we can come up with something that allows us to cleanly have both in a single example, but haven't yet come up with a good idea.

Reply:
Agree with @patrickbkr here.